src/Cli/dotnet/commands/dotnet-test/Logger.cs

```diff
@@ -30,6 +30,12 @@ public static void LogTrace(Func<string> messageLog)
 
                 lock (_lock)
                 {
+                    string directoryPath = Path.GetDirectoryName(_traceFilePath);
+                    if (!string.IsNullOrEmpty(directoryPath) && !Directory.Exists(directoryPath))
```

Comment:
I think Directory.CreateDirectory will internally check whether the directory exists.

```suggestion
                    if (!string.IsNullOrEmpty(directoryPath))
```